sdk/messaging/azservicebus/receiver.go

```diff
@@ -172,6 +172,159 @@ func (r *Receiver) ReceiveMessages(ctx context.Context, maxMessages int, options
 	return r.receiveMessagesImpl(ctx, maxMessages, options)
 }
 
+// ReceiveDeferredMessages receives messages that were deferred using `Receiver.DeferMessage`.
+func (r *Receiver) ReceiveDeferredMessages(ctx context.Context, sequenceNumbers []int64) ([]*ReceivedMessage, error) {
+	_, _, mgmt, _, err := r.amqpLinks.Get(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	amqpMessages, err := mgmt.ReceiveDeferred(ctx, r.receiveMode, sequenceNumbers)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var receivedMessages []*ReceivedMessage
+
+	for _, amqpMsg := range amqpMessages {
+		receivedMsg := newReceivedMessage(ctx, amqpMsg)
+		receivedMsg.deferred = true
+
+		receivedMessages = append(receivedMessages, receivedMsg)
+	}
+
+	return receivedMessages, nil
+}
+
+// PeekOptions contains options for the `Receiver.PeekMessages`
+// function.
+type PeekOptions struct {
```

Comment:
Nit: shouldn't this be `PeekMessagesOptions` per [guidelines](https://azure.github.io/azure-sdk/golang_introduction.html#optional-parameters). Are there multiple types of Peeks or is the only Peek PeekMessages?

Reply:
I think you're right, I'll rename.

Reply:
Renamed. Good catch!